### PR TITLE
Opacity feature

### PR DESCRIPTION
**Pull request by ismailuddin**
Added ability to change opacity for markers in functions:
- `px.scatter()`
- `px.scatter_3d()`
- `px.scatter_matrix()`
- `px.scatter_ternary()`
- `px.scatter_polar()`

## Usage
```
px.scatter(df, x='column1', y='column2', opacity=0.5)
```
Simply add the `opacity` argument to any of the functions, and specific a float between 0 and 1.

### CODE REVIEW

**Comment by gepcel** [comment]
There's `alpha` everywhere in matplotlib, is it possible to support `alpha` as an alias? .It's ok if it's not. I understand it's too more to ask.

**Comment by ismailuddin** [comment]
It's just because the way `opacity` is specified in `plotly` graphs is different for every type of plot, unlike `maptlotlib` where its usually a top level argument in the main function. So for the other types of graphs, I need to check specifically how `plotly` handles it for those graphs. I'll try to look into it when I have time.

**Comment by gepcel** [comment]
It's just matplotlib uses `alpha` in python, and ggplot uses `alpha` in R. So if `plotly_express` can support `alpha` as an alias name of `opacity`, it would be more easy for people not familiar with plotly.

**Comment by ismailuddin** [comment]
Oh sorry I misunderstood what you meant originally, yes I guess it could be done that way, but that's probably up to the `plotly_express` / `plotly` developers to decide.  Probably have to keep `opacity` for sake of consistency...

**Comment by nicolaskruchten** [comment]
Thanks for this PR! Our first contribution to the source :)

I do favour `opacity` instead of `alpha` as that's what it's called in the underlying tooling. `alpha` is very technical and not at all descriptive of the visual variable being controlled :)

A couple of requests:

1) Can you add a docstring in `plotly_express/_doc.py` please?
2) It would be best if you could set this in the `infer_config()` function as it's the same value for all points (the `make_trace_kwargs()` function is intended for attributes which vary from point to point)... this isn't at all clear from the code today and for this I apologize: I'll try to rename the functions and/or add some signpost comments to guide future PRs!

**Comment by ismailuddin** [comment]
You're welcome! I'm really glad this package got made, I was seriously thinking of making something similar myself the other day and now its already been done!!

I've moved the code as requested, and added the docstring. I had an error generating the HTML docs (`ValueError: I/O operation on closed file`) even prior to any modification, which I haven't quite figured out why though...

File: plotly_express/_core.py [hunk 1]
```diff
@@ -133,6 +133,8 @@ def make_trace_kwargs(args, trace_spec, g, mapping_labels, sizeref, color_range)
                     or len(args["data_frame"][name].unique()) <= 20
                 )
             ]
+        elif k == 'opacity':
```
Comment (nicolaskruchten) [inline plotly_express/_core.py]: oh you can remove this if you don't add opacity to `attrables`

File: plotly_express/_core.py [hunk 2]
```diff
@@ -568,7 +569,7 @@ def one_group(x):
 
 def infer_config(args, constructor, trace_patch):
     attrables = (
-        ["x", "y", "z", "a", "b", "c", "r", "theta", "size"]
+        ["x", "y", "z", "a", "b", "c", "r", "theta", "size", "opacity"]
```
Comment (nicolaskruchten) [inline plotly_express/_core.py]: this isn't necessary any more :)

File: plotly_express/_doc.py [hunk 3]
```diff
@@ -91,6 +91,11 @@
         "Ignored if `error_z` is `None`.",
     ],
     color=[colref, "Values from this column are used to assign color to marks."],
+    opacity=[
+        colref,
```
Comment (nicolaskruchten) [inline plotly_express/_doc.py]: in this case, opacity isn't a `colref` but a value between 0 and 1 actually, with a smart default.

File: plotly_express/_core.py (hunk 1 above)
Comment (ismailuddin) [reply plotly_express/_core.py]: If I remove this, it attempts to use the value of opacity as a column name to find in the dataframe and set according to it's values... so I'm not sure how else to go about it?
```
KeyError: 0.1
...
    228             else:
    229                 if v:
--> 230                     result[k] = g[v]
    231                 mapping_labels.append(("%s=%%{%s}" % (v_label, k), None))
    232     if trace_spec.constructor not in [
```

Comment (ismailuddin) [reply plotly_express/_core.py]: So it was actually pretty simple haha...

File: plotly_express/_core.py [hunk 4]
```diff
@@ -620,6 +619,8 @@ def infer_config(args, constructor, trace_patch):
         grouped_attrs.append("marker.symbol")
 
     trace_patch = trace_patch.copy()
+    if "opacity" in args:
```
Comment (nicolaskruchten) [inline plotly_express/_core.py]: probably needs a `None` check also :)

**Comment by nicolaskruchten** [comment]
Yep, pretty simple in the end :)

File: plotly_express/_core.py (hunk 4 above)
Comment (nicolaskruchten) [reply plotly_express/_core.py]: actually not even. the none will just be ignored downstream

File: plotly_express/_doc.py [hunk 5]
```diff
@@ -91,6 +91,10 @@
         "Ignored if `error_z` is `None`.",
     ],
     color=[colref, "Values from this column are used to assign color to marks."],
+    opacity=[
+        "Sets the opacity of the markers in scatter plots such as `scatter()`",
```
Comment (nicolaskruchten) [inline plotly_express/_doc.py]: I'll tweak this wording a bit. To match the rest it should be something like `(number between 0 and 1) Sets the opacity of markers.` (no need to enumerate the function types as this shows up in the individual doc strings of those functions.)

**Comment by nicolaskruchten** [comment]
I'm happy to merge this in a few hours when I've had a chance to test it a bit. Thanks again for the contribution and feel free to suggest additional ones now that you've seen the pattern :)

File: plotly_express/_doc.py (hunk 5 above)
Comment (ismailuddin) [reply plotly_express/_doc.py]: I've adjusted to keep it more inline, feel free to change as necessary!

**Comment by nicolaskruchten** [comment]
Thanks again!

**Comment by DmitriyG228** [comment]
Hi, scatter_mapbox and scatter_geo  also seen to miss opacity feature

**Comment by nicolaskruchten** [comment]
@DmitriyG228 thanks for letting me know! I'll add this to the next release, which should come out within the hour :)

**Pull request merged**